### PR TITLE
debug: add button in admin panel to login to other accounts

### PR DESCRIPTION
**Pull request by justcool393**
should be mostly ready to go, just need to test

### CODE REVIEW

**Comment by justcool393** [comment]
tested and it works. yay!

File: files/helpers/wrappers.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def get_logged_in_user():
 		if lo_user:
 			id = int(lo_user)
 			v = g.db.query(User).get(id)
+			print(v)
```
Comment (official-techsupport) [inline files/helpers/wrappers.py]: Remove this probably?

Comment (justcool393) [reply files/helpers/wrappers.py]: done

File: files/routes/__init__.py [hunk 2]
```diff
@@ -14,4 +14,5 @@
 from .feeds import *
 from .awards import *
 from .giphy import *
+from .dev import *
```
Comment (TLSM) [inline files/routes/__init__.py]: Any way to have stronger protections to keep these routes out of production? Conditional import? Or a `@debug_only` decorator just after the routing decorator on all the routes?

File: files/__main__.py [hunk 3]
```diff
@@ -170,6 +170,7 @@ def before_request():
 		return 'Please use a "User-Agent" header!', 403
 
 	ua = g.agent.lower()
+	g.debug = bool_from_string(app.config["DEBUG"])
```
Comment (TLSM) [inline files/__main__.py]: We use `app.debug`, `app.config["DEBUG"]`, and `g.debug` all essentially interchangeably. Perhaps use `app.debug` here to be consistent with L97 and standardize around the `{app,g}.debug` variants.

**Review by TLSM — CHANGES_REQUESTED** [review]

File: files/routes/__init__.py (hunk 2 above)
Comment (justcool393) [reply files/routes/__init__.py]: need to test if this causes import weirdness (no importmania on this fork 😔) but would prefer the conditional import if at all possible tbh

**Comment by justcool393** [comment]
done @TLSM :)

**Review by TLSM — APPROVED** [review]

**Comment by zorbathut** [comment]
More conflicts, sorry to say :( Looks fine though!

File: files/routes/dev.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+@app.post('/dev/sessions/')
+@admin_level_required(PERMS['DEBUG_LOGIN_TO_OTHERS'])
+def login_to_other_account(v):
+	u = get_user(request.values.get('username'))
+	session.permanent = True
+	session["lo_user"] = u.id
+	session["login_nonce"] = u.login_nonce
+	session["session_id"] = token_hex(49)
+	return redirect('/')
```
Comment (zorbathut) [inline files/routes/dev.py]: I'm worried that this is guarded from release builds *only* by the nonexistence of an import, which seems like something that could be accidentally imported from somewhere else and we wouldn't realize it. Could I get a `if not app.debug: return` at the top, or something like that?

**Review by zorbathut — CHANGES_REQUESTED** [review]

File: files/routes/dev.py (hunk 4 above)
Comment (justcool393) [reply files/routes/dev.py]: :marseythumbsup:

**Review by zorbathut — APPROVED** [review]

**Pull request merged**